### PR TITLE
Mat 5099 harp id case insensitive

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-5099](https://jira.cms.gov/browse/MAT-5099)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/EditTestCase.tsx [hunk 1]
```diff
@@ -292,13 +286,7 @@ const EditTestCase = () => {
...
         setTestCase(_.cloneDeep(tc));
         setEditorVal(tc.json ? tc.json : "");
-        setCanEdit(
-          measure?.createdBy === userName ||
-            measure?.acls?.some(
-              (acl) =>
-                acl.userId === userName && acl.roles.indexOf("SHARED_WITH") >= 0
-            )
-        );
+        setCanEdit(canEdit);
```
Comment (nmorasb) [inline src/components/editTestCase/EditTestCase.tsx]: This seems unnecessary. Isn't it just setting the state variable "canEdit" to itself?

File: src/components/editTestCase/EditTestCase.tsx [hunk 2]
```diff
@@ -212,11 +210,7 @@ const EditTestCase = () => {
   const { updateMeasure } = measureStore;
   const load = useRef(0);
   const [canEdit, setCanEdit] = useState<boolean>(
-    measure?.createdBy === userName ||
-      measure?.acls?.some(
-        (acl) =>
-          acl.userId === userName && acl.roles.indexOf("SHARED_WITH") >= 0
-      )
+    checkUserCanEdit(measure?.createdBy, measure?.acls)
```
Comment (nmorasb) [inline src/components/editTestCase/EditTestCase.tsx:213]: this logic seems incorrect...shouldn't we be checking that the currently logged in user has edit permissions? This appears to be checking only if the measure creator has edit permissions.

File: src/components/testCaseList/TestCaseList.tsx [hunk 3]
```diff
@@ -89,14 +87,8 @@ const TestCaseList = () => {
...
   useEffect(() => {
-    setCanEdit(
-      measure?.createdBy === userName ||
-        measure?.acls?.some(
-          (acl) =>
-            acl.userId === userName && acl.roles.indexOf("SHARED_WITH") >= 0
-        )
-    );
-  }, [measure, userName]);
+    setCanEdit(checkUserCanEdit(measure?.createdBy, measure?.acls));
```
Comment (nmorasb) [inline src/components/testCaseList/TestCaseList.tsx:90]: same as on EditTestCase - this seems like we're checking if the measure creator has edit permissions. Shouldn't we check against the currently logged in user?

File: src/components/editTestCase/EditTestCase.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/EditTestCase.tsx:213]: @nmorasb Yes we are checking the currently logged in user has edit permissions. That part of logic is done in madie-util, which takes the logged in user from okta, and does the comparison with the passed in createdBy or acls.

File: src/components/testCaseList/TestCaseList.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCaseList.tsx:90]: @nmorasb Same as above, thanks so much Nate.

Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCaseList.tsx:90]: @nmorasb the logic is here for your reference: https://github.com/MeasureAuthoringTool/madie-util/pull/25/files

File: src/components/editTestCase/EditTestCase.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/EditTestCase.tsx]: @nmorasb yeah you are right removed it thanks

File: src/components/editTestCase/EditTestCase.tsx (hunk 2 above)
Comment (nmorasb) [reply src/components/editTestCase/EditTestCase.tsx:213]: Ah! Ok gotcha, that makes sense. Sorry, wasn't familiar with the logic inside that function and was just inferring from the function name and parameters!

File: src/components/testCaseList/TestCaseList.tsx (hunk 3 above)
Comment (nmorasb) [reply src/components/testCaseList/TestCaseList.tsx:90]: gotcha

**Review by nmorasb — APPROVED** [review]

**Pull request merged**